components/browser/session/src/main/java/mozilla/components/browser/session/SessionObserver.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.session
+
+/**
+ * This class is a combination of [Session.Observer] and
+ * [SessionManager.Observer]. It provides functionality to observe changes to a
+ * specified or selected session, and automatically takes care of switching
+ * over the observer in case a different session is selected (see [observe] and
+ * [observeSelected]).
+ *
+ * @property activeSession the currently observed session
+ * @property sessionManager the application's session manager
+ */
+abstract class SessionObserver(
+    private val sessionManager: SessionManager
+) : SessionManager.Observer, Session.Observer {
+
+    protected open var activeSession: Session? = null
+
+    /**
+     * Starts observing changes to the specified session.
+     *
+     * @param session the session to observe.
+     */
+    fun observe(session: Session) {
```

Comment:
API nit: Would it be even more obvious if we name this something like `observeOnly()` or `observeAlways`?